azure/scope/managedcontrolplane.go

```diff
@@ -254,6 +254,7 @@ func (s *ManagedControlPlaneScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			VNetName:          s.Vnet().Name,
 			VNetResourceGroup: s.Vnet().ResourceGroup,
 			IsVNetManaged:     s.IsVnetManaged(),
+			IsSubnetManaged:   s.Vnet().Tags.HasOwned(s.ClusterName()),
```

Comment:
unfortunately this won't work for managed clusters because `.Vnet()` [returns a new Vnet](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2522/files#diff-cfd94adbc2ff767742e457e64836e93e88c6e03c3a64c04619c54b7ed76f284eR199) with no tags set since the AzureManagedControlPlane spec doesn't actually have a Vnet spec that is persisted, so this will return `false` every time. The only way to is to check the actual tags on the vnet by calling the virtualnetworks service `IsManaged` function.

Reply:
Okay, I'll go back to the previous approach and see if I can spend some more time trying to figure out how to properly mock and pass around the virtualnetworks service & scope while in the tests for the subnets. But TBH it kind of feels like that's crossing some boundary lines by doing that. I'll noodle on it and see what I come up with...

Reply:
Yeah it's definitely not ideal 😕 Happy to brainstorm in a pairing session next week if that helps. 

In other news I don't know if you saw my [comment](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2525#issuecomment-1207082254 ) in the other PR but the alternative strategy of not deleting subnets at all might not actually work out because Azure isn't happy about route tables attached to a subnet being deleted before the subnet (and the route table has to be deleted before the vnet so can know if it's managed) so this complicates things quite a bit. This current PR might be our best way forward.

Reply:
> Happy to brainstorm in a pairing session next week if that helps.

That might be very helpful. Thank you. I'll let you know where my brain ends up as it simmers over the weekend.

> the alternative strategy of not deleting subnets at all might not actually work out

yeah, I saw that. I was originally going to suggest that we go with that approach instead but then I saw the other comment and got all :( face.

This should be doable, it's more my limited experience with mocking things in Go than anything else.